File: 0739-daily-temperatures/0739-daily-temperatures.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        int n = temperatures.length;
        int[] result = new int[n];
        Stack<Integer> indStack = new Stack<Integer>();
        
        for(int i=0; i<n; i++){
            while(!indStack.isEmpty() && temperatures[i] > temperatures[indStack.peek()]){
                result[indStack.peek()] = i - indStack.pop();
            }
            indStack.push(i);
        }
        
        return result;
        
        
    }
}

File: 0739-daily-temperatures/0739-daily-temperatures.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        
        vector<int> rv;
        stack<int> s;
        
        rv.resize(temperatures.size());
        
        for(int i=0; i<temperatures.size(); i++){
            while(s.size() && temperatures[i] > temperatures[s.top()]){
                rv[s.top()] = i - s.top();
                s.pop();
            }
            s.push(i);
        }
        
        return rv;
        
    }
};